### ISSUE TITLE
Explicit "content-length" in header leads to incorrect HTTP request

### ISSUE THREAD

**Issue by k3kc** [scrapy/scrapy#4919]
### Description

Explicit "content-length" in header leads to incorrect HTTP request - 400 error from target

### Steps to Reproduce
```
Request("https://webhook.site/<YOUR ID>", method='POST', headers={}) >>> OK

Request("https://webhook.site/<YOUR ID>", method='POST', headers={'content-length': '0'}) >>> ERROR: <twisted.python.failure.Failure scrapy.spidermiddlewares.httperror.HttpError: Ignoring non-200 response>
```

**Expected behavior:** [What you expect to happen]
`OK`

**Actual behavior:** [What actually happens]
`ERROR`

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

### Versions
2.4.1

### Additional context
Seems twisted duplicates content-length field in the resulting request

**Comment by mmitropoulou** [comment]
Hello @wRAR! @marlenachatzigrigoriou and I would like to contribute to this issue. We are new to scrapy so we would like to have your guidance. Thank you!

**Comment by mmitropoulou** [comment]
Hello, I have tried to recreate this issue. I have tried in my version of scrapy (2.5.0) and in the proposed here (2.4.1), but the result is fine and no problem occures .Can you please help me understand if im doing something wrong? This is the spider that I run:
def start_requests(self):
     yield scrapy.Request('https://webhook.site/thecode’ method='POST', headers={'content-length': '0'})

**Comment by wRAR** [comment]
I've actually just got a 400 response with this spider.

**Comment by mmitropoulou** [comment]
Me too. In order to solve this i added in my code the following line:
handle_httpstatus_list = [400]
but still no problem like the one described above.

**Comment by wRAR** [comment]
The problem is receiving the 400 response.

**Comment by Gallaecio** [comment]
I am not sure what the best approach is.

I mean, ideally the user should not set the `Content-Length` header at all, and let Scrapy (Twisted) do that.

If a user sets the header, we could drop it from the download handlers to avoid this issue. However, doing so silently could be confusing. For example, imagine the user sets the wrong header value. Do we ignore the user-defined value without any feedback?

Maybe we should error out on the client side (i.e. do not let the request reach the target server) on requests with the header, and ask the user to remove the header. And if someone has a valid use case for setting `Content-Length` to the _wrong_ value, let them explain and defend that use case in a GitHub feature request.

**Comment by wRAR** [comment]
Yeah, an error looks better to me than just ignoring the user-set header.

**Comment by andreastziortz** [comment]
@Gallaecio I do agree that the user should not be able to add a header. The only other possible route would be to let user overwrite Content-Length if included in headers, if that's a valuable option. But I see the aforementioned suggestion as a better approach, as it clearly communicated the expected behaviour and set remove from user complexity that might confuse them.
